### PR TITLE
FIX: symbols and strings should be treated similarly while saving custom fields

### PR DESCRIPTION
**Pull request by fzngagan**
- As discussed here
https://meta.discourse.org/t/inconsistent-behaviour-in-userupdater-class/161360

### CODE REVIEW

**Comment by fzngagan** [comment]
- I see some checks are failing but I haven't touched the frontend code I promise ;)

**Comment by eviltrout** [comment]
Hey @fzngagan this looks good but we will need a test to confirm the functionality and to avoid regressions.

**Comment by fzngagan** [comment]
@eviltrout I uncovered an issue and accounted for all different cases.

File: spec/components/concern/has_custom_fields_spec.rb [hunk 1]
```diff
@@ -337,6 +337,32 @@ class CustomFieldsTestItemCustomField < ActiveRecord::Base
         expect(test_item.custom_fields['hello']).to eq('world')
         expect(test_item.custom_fields['abc']).to eq('ghi')
       end
+
+      it 'allows using string and symbol indices interchangably' do
```
Comment (ZogStriP) [inline spec/components/concern/has_custom_fields_spec.rb:341]: Small typo

```suggestion
      it 'allows using string and symbol indices interchangeably' do
```

File: spec/components/concern/has_custom_fields_spec.rb [hunk 2]
```diff
@@ -337,6 +337,32 @@ class CustomFieldsTestItemCustomField < ActiveRecord::Base
...
+         expect(test_item.custom_fields[:bob]).to eq('marley')
+         expect(test_item.custom_fields[:jack]).to eq('black')
+
+         # Persisted
+         test_item.save
+         test_item.reload
+         expect(test_item.custom_fields[:bob]).to eq('marley')
+         expect(test_item.custom_fields[:jack]).to eq('black')
+
+         ## Update via string index again
```
Comment (ZogStriP) [inline spec/components/concern/has_custom_fields_spec.rb:357]:
```suggestion
         # Update via string index again
```

**Comment by fzngagan** [comment]
@ZogStriP thanks for pointing out the typos. Accepted both the changes 👍

**Comment by eviltrout** [comment]
Thank you!

**Comment by fzngagan** [comment]
@eviltrout I think you didn't squash merge as always but that's fine I guess? :)

**Comment by sbrighiu** [comment]
regression :( https://meta.discourse.org/t/custom-fields-called-simultaneous-cause-value-to-turn-into-array/162034/2

**Comment by eviltrout** [comment]
@sbrighiu I don't see how this is a regression?

**Comment by sbrighiu** [comment]
Hi @eviltrout,

It seems to be related from what @fzngagan is saying.

All I know is I call the same endpoint 3-8 at the same time and when I check the custom fields values, they are Arrays instead of Strings.

**Pull request merged**